waku/v2/node/wakuoptions.go

```diff
@@ -246,6 +246,14 @@ func WithWakuRelayAndMinPeers(minRelayPeersToPublish int, opts ...pubsub.Option)
 	}
 }
 
+// WithoutWakuRelay disables the Waku V2 Relay protocol.
+func WithoutWakuRelay() WakuNodeOption {
```

Comment:
Why do we want this? Isn't a node without relay kinda useless?

Reply:
A node without relay is useful in the case where you just want a node client that does things through `filter` and `store`. The use we have that surfaced these leaks is actually one of those where we could have just disabled relay to avoid them, but I wanted to fix them even if we'll just disable relay via this anyway.